devito/finite_differences/derivative.py

```diff
@@ -277,8 +286,25 @@ def evaluate(self):
         # types of discretizations.
         return self._eval_fd(self.expr)
 
+    @property
+    def _eval_deriv(self):
+        return self._eval_fd(self.expr)
+
     def _eval_fd(self, expr):
-        expr = getattr(expr, 'evaluate', expr)
+        """
+        Evaluate finite difference approximation of the Derivative.
+        The evaluation goes in four steps:
```

Comment:
'goes' doesn't really fit here, maybe `Evaluation is carried out via the following four steps:`?